lib/i18n/backend/discourse_i18n.rb

```diff
@@ -33,6 +35,13 @@ def pluralize(locale, entry, count)
         end
       end
 
+      def self.sort_locale_files(files)
+        files.sort_by do |filename|
+          matches = /(client|server)-([1-9]|[1-9][0-9]|100)\.([a-z]{2})(\_[A-Z]{2})?\.yml/.match(filename)
+          matches&.[](2)&.to_i || 0
```

Comment:
I suggest removing the locale from the regex. Even though all locales shipped with Discourse adhere to that format, it's definitely not a requirement.

```suggestion
          matches = /(?:client|server)-([1-9]|[1-9][0-9]|100)\..+\.yml/.match(filename)
          matches&.[](1)&.to_i || 0
```